src/main/scala/Joint.scala

```diff
@@ -47,14 +113,28 @@ object Joint {
   * @param shape A list of lines specifying the shape of the joint. Each item is a
   * 3-tuple. The first two items specify the line, while the last gives the distance
   * of the line from the joint's center in the local coordinate system.
+  * @param dipAngleParam An optional parameter that can be used to specify the dip angle for the joint.
+  *                 This avoids recalculation of a known dip angle.
+  * @param dipDirectionParam An optional parameter that can be used to specify the dip direction for
+  *                     the joint. This avoids recalculation of a known dip direction.
   */
 case class Joint(normalVec: (Double, Double, Double), localOrigin: (Double, Double, Double),
-                 center: (Double, Double, Double), dipAngle: Double, dipDirection: Double, phi: Double,
-                 cohesion: Double, shape: Seq[((Double, Double, Double),Double)]) {
+                 center: (Double, Double, Double), phi: Double, cohesion: Double,
+                 shape: Seq[((Double, Double, Double),Double)],
+                 dipAngleParam: Option[Double]=null,
```

Comment:
I tried this, but it gives a type mismatch: found Null(null), required Double. Seems like it needs Option[Double] since null isn't the same type as Double. Am I missing something here? Based on what I could find online, it seems like the whole option thing is there to avoid having to do null checks.